plugins/outputs/datadog/datadog.go

```diff
@@ -136,17 +122,44 @@ func (d *Datadog) authenticatedUrl() string {
 	return fmt.Sprintf("%s?%s", d.apiUrl, q.Encode())
 }
 
-func buildMetrics(m telegraf.Metric) (map[string]Point, error) {
-	ms := make(map[string]Point)
-	for k, v := range m.Fields() {
-		var p Point
-		if err := p.setValue(v); err != nil {
-			return ms, fmt.Errorf("unable to extract value from Fields, %s", err.Error())
+// Convert a telegraf metric to datadog metrics;
+// we need a separate metric for each field.
+// (also has magic for statsd field names)
+func buildMetrics(m telegraf.Metric) ([]*Metric, error) {
+	var datadogMetrics []*Metric
+	tags := m.Tags()
+	metricType := tags["metric_type"]
+	baseDatadogMetric := Metric{
+		Metric:   m.Name(),
+		Tags:     buildTags(tags),
+		Host:     tags["host"],
+		Interval: float64(m.Interval()) / float64(time.Second),
+	}
+
+	for field, value := range m.Fields() {
+		metric := baseDatadogMetric
+		if field != "value" {
+			metric.Metric += "_" + field
 		}
+		metric.Metric = strings.Replace(metric.Metric, "_", ".", -1)
+		p := &metric.Points[0]
 		p[0] = float64(m.Time().Unix())
-		ms[k] = p
+		if err := p.setValue(value); err != nil {
+			return nil, fmt.Errorf("unable to extract value from Fields, %s", err.Error())
+		}
+
+		if metricType == "counter" ||
```

Comment:
This is consistent with the dogstatsd behaviour. Admittedly, datadog also supports a (similarly undocumented) 'counter' type, but the dogstatsd server doesn't use it. But I definitely should make a comment to this effect.